extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/widgets/CompassWidget.kt

```diff
@@ -32,4 +33,5 @@ class CompassWidget(
   position = position,
   marginX = marginX,
   marginY = marginY,
+  listener = BitmapWidgetListener { it.recycle() }
```

Comment:
```suggestion
```

If the current approach is desired this will create a semver incompatibility and delay any potential stable releases

I think you should prefer solutions that would not cause semver. For example, the `BitmapWidget` is already an open class so why not give the ability to override the functionality there??

``` kotlin
@MapboxExperimental
open class BitmapWidget @JvmOverloads constructor(
  bitmap: Bitmap,
  position: WidgetPosition = WidgetPosition(
    vertical = WidgetPosition.Vertical.TOP,
    horizontal = WidgetPosition.Horizontal.LEFT,
  ),
  marginX: Float = 0f,
  marginY: Float = 0f,
) : Widget() {
  override val renderer = BitmapWidgetRenderer(
    bitmapWidget = this, // <<< now the renderer can call onLoaded
    bitmap = bitmap,
    position = position,
    marginX = marginX,
    marginY = marginY,
  )

  // This is called when the BitmapWidgetRenderer is done with the bitmap
  fun onLoaded() {
    bitmap.recycle()
  }

  /**
   * Update bitmap widget uses.
   */
  fun updateBitmap(bitmap: Bitmap) {
    renderer.updateBitmap(bitmap)
  }

  override fun setTranslation(translationX: Float, translationY: Float) {
    renderer.setTranslation(translationX = translationX, translationY = translationY)
  }

  override fun setRotation(angleDegrees: Float) {
    renderer.setRotation(angleDegrees = angleDegrees)
  }
}
```

Reply:
This is why we really need to merge this first https://github.com/mapbox/mapbox-maps-android/pull/1706

That way the incompatibility is caught by the build system

Reply:
Hi @kmadsen, since you mention that the class is `open`, it got me thinking whether we implement the `onLoaded` as you suggest, but have a subclassed `RecycledBitmapWidget` which calls `bitmap.recycle` in its `onLoaded`.  The user is then free to decide which class to use. Our own `CompassWidget` and `LogoWidget` could inherit from `RecycledBitmapWidget`. Would this be acceptable?

Reply:
Yeah this is better with a `onLoaded` open function in my opinion.

`RecycledBitmapWidget` could have been good if it was the original implementation, unfortunately the bug fix also requires a code change. So technically will not use `RecycledBitmapWidget` until it becomes part of a final release like `10.9.0` or `10.10.0`

If you want to make it so this is a "free fix upon upgrade". Then make it part of the `BitmapWidget`